TicTacToeProject/Board.cs

```diff
@@ -17,32 +17,53 @@ public Board()
             }
         }
 
-
-        public bool isEmpty()
+        public bool isBoardEmpty()
         {
             return board.TrueForAll(space => space.isSpaceEmpty());
         }
 
-
-        public bool isFilled()
+        public bool isBoardFilled()
         {
             return board.TrueForAll(space => !space.isSpaceEmpty());
         }
 
+        public void placeMarker(int location, string playerMarker) 
+        {
+            board[location - 1].marker = playerMarker;
+        }
+
+        public string markerAtLocation(int location) 
+        {
+            return board[location - 1].marker;
+        }
+
+        public bool isSpaceOnBoardEmpty(int location)
+        {
+            return board[location - 1].isSpaceEmpty();
+        }
 
-        public bool placeMarker(int location, string playerMarker) 
+        public List<int> getAvailableSpaces()
         {
-            if (board[location - 1].isSpaceEmpty())
+            var availableSpaces = new List<int>();
```

Comment:
LINQ is pretty nice in these kinds of cases. This method could look like this instead:

```
            return board
                .Where(space => space.isSpaceEmpty())
                .Select(space => space.location)
                .ToList();
```